### PR TITLE
Firebase login

### PR DESCRIPTION
**Pull request by chris-pica**

### CODE REVIEW

File: src/lib/components/auth/FirebaseLogin.svelte [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+			var uiConfig = {
+				callbacks: {
+					signInSuccessWithAuthResult: function(authResult, redirectUrl) {
+						return true; // redirect automatically
+					},
+					// signInFailure: function(error) {
+					// 	// TODO https://github.com/firebase/FirebaseUI-Web#signinfailureerror
+					// 	console.error(error);
+					// },
```
Comment (Kansuler) [inline src/lib/components/auth/FirebaseLogin.svelte:19]: Perhaps to be removed?

File: src/lib/components/auth/Logout.svelte [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<script lang="ts">
+	import { goto } from '$app/navigation';
+	import { getAuth } from "firebase/auth";
+
+	function logout() {
+		getAuth().signOut().then(function() {
+			goto('/');
+		}, function(error) {
+			// TODO
```
Comment (Kansuler) [inline src/lib/components/auth/Logout.svelte:9]: What Todo?

File: src/lib/components/auth/FirebaseLogin.svelte [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+						// hide the loader when widget is rendered
+						document.getElementById('firebaseui-loader').style.display = 'none';
+					}
+				},
+				signInSuccessUrl: '/',
+				signInOptions: [
+					firebase.auth.EmailAuthProvider.PROVIDER_ID,
+				],
+				// TODO
+				// privacyPolicyUrl: '<your-tos-url>'
```
Comment (Kansuler) [inline src/lib/components/auth/FirebaseLogin.svelte:30]: What todo?

**Review by Kansuler — COMMENTED** [review]
Just a few nitpicks, looks good! 👍

File: src/lib/components/auth/FirebaseLogin.svelte (hunk 1 above)
Comment (chris-pica) [reply src/lib/components/auth/FirebaseLogin.svelte:19]: I think we need to gracefully handle some failures, add a todo?

File: src/lib/components/auth/FirebaseLogin.svelte (hunk 3 above)
Comment (chris-pica) [reply src/lib/components/auth/FirebaseLogin.svelte]: Clarified

File: src/lib/components/auth/Logout.svelte (hunk 2 above)
Comment (chris-pica) [reply src/lib/components/auth/Logout.svelte]: Clarified

**Review by varun-pica — APPROVED** [review]

**Comment by Kansuler** [comment]
LGTM

**Review by Kansuler — APPROVED** [review]

**Pull request merged**